### PR TITLE
Special security announcement may

### PR DESCRIPTION
**Pull request by anjai94**
Adding following advisory into the documentation page.  

- WSO2-2022-2177
- WSO2-2022-2182

### CODE REVIEW

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    ......
+</ResourceAccessControl>
+```
+
+If you have already applied the above given configuration in your identity.xml, please add the additional `<Permissions>/</Permissions>`  line as mentioned above & restart the server.
+
+
+### CREDITS
+WSO2 thanks, **[Rick Roane](https://www.linkedin.com/in/rick-r/)** for responsibly reporting the identified issue and working with us as we addressed it.
+
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:159]: Let's change, 
```
[^1] [https://is.docs.wso2.com/en/latest/guides/password-mgt/recover-password/ ](https://is.docs.wso2.com/en/latest/guides/password-mgt/recover-password/)<br>
[^2] [https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/user-account-management/#enable-password-recovery ](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/user-account-management/#enable-password-recovery) <br>
[^3] [https://ob.docs.wso2.com/en/latest/get-started/set-up-accelerators/ ](https://ob.docs.wso2.com/en/latest/get-started/set-up-accelerators/)
```
to,

```
### References
[^1]: [https://is.docs.wso2.com/en/latest/guides/password-mgt/recover-password/ ](https://is.docs.wso2.com/en/latest/guides/password-mgt/recover-password/)
[^2]: [https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/user-account-management/#enable-password-recovery ](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/user-account-management/#enable-password-recovery)
[^3]: [https://ob.docs.wso2.com/en/latest/get-started/set-up-accelerators/ ](https://ob.docs.wso2.com/en/latest/get-started/set-up-accelerators/)
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    <Resource context="/" secured="false" http-method="GET"/>
+    ......
+</ResourceAccessControl>
+```
+
+If you have already applied the above given configuration in your identity.xml, please add the additional `<Permissions>/</Permissions>`  line as mentioned above & restart the server.
+
+
+### CREDITS
+WSO2 thanks, **[Rick Roane](https://www.linkedin.com/in/rick-r/)** for responsibly reporting the identified issue and working with us as we addressed it.
```
[the remark this answers is not in the document]
Comment (1uffyD9) [reply en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:158]: Let's check the same for other contents.

**Review by 1uffyD9 — CHANGES_REQUESTED** [review]
Have suggested few changes.

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2101.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+### SOLUTION
+If the latest version of the affected WSO2 product is not mentioned under the affected product list, you may migrate to the latest version to receive security fixes. Otherwise you may apply the relevant fixes to the product based on the public fix: 
+
+* [https://github.com/wso2-extensions/identity-governance/pull/616](https://github.com/wso2-extensions/identity-governance/pull/616)
+
+Further, If you are not using above mentioned behavior intentionally, please follow the below steps:
+
+* Apply the provided patch/update to the affected versions of the products.
+* Add the following configuration to deployment.toml file.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2101.md:44]:
```suggestion
* Add the following configuration to `deployment.toml` file.
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2101.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    `[event.default_listener.governance_identity_store] enable_hybrid_data_store = false`
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2101.md:45]:
```suggestion
    ```toml
    [event.default_listener.governance_identity_store] enable_hybrid_data_store = false
    ```
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+You may migrate to the latest version of the product if the latest version is not listed under the list of the affected products. Otherwise, you may apply the relevant fixes to the product based on the public fixes as given below:
+
+* [https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/216](https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/216)
+
+Or else you may follow the mitigation steps given below.
+
+#### Group 1:
+For below listed products, it is reoccomand to append the following configuration lines to the "&lt;PRODUCT-HOME&gt;/repository/conf/deployment.toml" file and restart the server.
+
+
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:107]:
```suggestion
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Identity Server 5.9.0, 5.10.0, 5.11.0, 6.0.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:108]:
```suggestion
* WSO2 Identity Server 5.9.0, 5.10.0, 5.11.0, 6.0.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Identity Server as Key Manager 5.9.0, 5.10.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:109]:
```suggestion
* WSO2 Identity Server as Key Manager 5.9.0, 5.10.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 API Manager 3.0.0, 3.1.0, 3.2.0, 4.0.0, 4.1.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:110]:
```suggestion
* WSO2 API Manager 3.0.0, 3.1.0, 3.2.0, 4.0.0, 4.1.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Open Banking AM 2.0.0, 3.0.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:111]:
```suggestion
* WSO2 Open Banking AM 2.0.0, 3.0.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Open Banking IAM 2.0.0, 3.0.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:112]:
```suggestion
* WSO2 Open Banking IAM 2.0.0, 3.0.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+
+```toml
+[[resource.access_control]]
+context="(.*)/((\\.+)|(.*;+.*)|%2e)/(.*)"
+permissions = ["/"]
+secure=true
+http_method="all"
+``` 
+If you have already applied the similar configuration in your "deployment.toml", you can add the above given configuration as a new config & restart the server.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:122]:
```suggestion
If you have already applied the similar configuration in your `deployment.toml`, you can add the above given configuration as a new config & restart the server.
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+#### Group 2:
+For group 2 products, please Add the below given config as the first entry of the `<ResourceAccessControl>` tag of "`<PRODUCT-HOME>`/repository/conf/identity/identity.xml" file and restart the server.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:125]:
```suggestion
For group 2 products, please Add the below given config as the first entry of the `<ResourceAccessControl>` tag of `<PRODUCT-HOME>/repository/conf/identity/identity.xml` file and restart the server.
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+
+- WSO2 Identity Server 5.3.0, 5.4.0, 5.4.1, 5.5.0, 5.6.0, 5.7.0, 5.8.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:128]:
```suggestion
* WSO2 Identity Server 5.3.0, 5.4.0, 5.4.1, 5.5.0, 5.6.0, 5.7.0, 5.8.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Identity Server as Key Manager 5.3.0, 5.5.0, 5.6.0, 5.7.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:129]:
```suggestion
* WSO2 Identity Server as Key Manager 5.3.0, 5.5.0, 5.6.0, 5.7.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Open Banking AM 1.3.0, 1.4.0, 1.5.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:130]:
```suggestion
* WSO2 Open Banking AM 1.3.0, 1.4.0, 1.5.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+- WSO2 Open Banking IAM 1.3.0, 1.4.0, 1.5.0
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:131]:
```suggestion
* WSO2 Open Banking IAM 1.3.0, 1.4.0, 1.5.0
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+
+```xml
+<Resource context="(.*)/((\.+)|(.*;+.*)|%2e)/(.*)" secured="true" http-method="all">
+            <Permissions>/</Permissions>
+</Resource>
+```
+
+Example:
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:139]: Shall we use the example code segment like this? 

```suggestion
!!! example
     ```xml
      <ResourceAccessControl>
          <Resource context="(.*)/((\.+)|(.*;+.*)|%2e)/(.*)" secured="true" http-method="all">
                 <Permissions>/</Permissions>
          </Resource>
          <Resource context="(.*)" secured="false" http-method="OPTIONS"/>
          <Resource context="/" secured="false" http-method="GET"/>
          ......
      </ResourceAccessControl>
      ```
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+    <Resource context="/" secured="false" http-method="GET"/>
+    ......
+</ResourceAccessControl>
+```
+
+If you have already applied the above given configuration in your identity.xml, please add the additional `<Permissions>/</Permissions>` line as mentioned above & restart the server.
+
+
+### CREDITS
+WSO2 thanks, **[Rick Roane]**(https://www.linkedin.com/in/rick-r/) for responsibly reporting the identified issue and working with us as we addressed it.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2177.md:156]:
```suggestion
WSO2 thanks, [Rick Roane](https://www.linkedin.com/in/rick-r/) for responsibly reporting the identified issue and working with us as we addressed it.
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2182.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+| WSO2 Open Banking IAM               |           2.0.0 |          208 | 1664976509070 |
+| WSO2 Open Banking IAM Accelerator   |           3.0.0 |           Refer to WSO2IAM-5.11.0 update level since this vulnerability is originating from the base product | N/A |
+
+
+If you are an open-source user or using a product version that is EOL (End of License) :
+
+You may migrate to the latest version of the product if the latest version is not listed under the list of the affected products. Otherwise, you may apply the relevant fixes to the product based on the public fixes as given below:
+
+
+- [https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1603](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1603)
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2182.md:73]:
```suggestion
* [https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1603](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1603) 
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2182.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+| WSO2 Open Banking AM Accelerator    |           3.0.0 |           Refer to WSO2AM-4.0.0 update level since this vulnerability is originating from the base product | N/A |
+| WSO2 Open Banking IAM               |           2.0.0 |          208 | 1664976509070 |
+| WSO2 Open Banking IAM Accelerator   |           3.0.0 |           Refer to WSO2IAM-5.11.0 update level since this vulnerability is originating from the base product | N/A |
+
+
+If you are an open-source user or using a product version that is EOL (End of License) :
+
+You may migrate to the latest version of the product if the latest version is not listed under the list of the affected products. Otherwise, you may apply the relevant fixes to the product based on the public fixes as given below:
+
+
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2182.md:72]:
```suggestion
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2022-2357.md [hunk 21]
```diff
@@ -26,10 +26,10 @@ The **Client ID** and **Client Secret** values of the connection configuration t
...
+There is no security impact for your environment even though the client secret was logged into the carbon audit logs However, as an additional precaution, we highly recommend rotating IDP client secrets that are being used in your environment after applying the given patch. The following two approaches can be recommended for IDP secret rotation.
 
 * If the application supports the maintenance of multiple secrets, then update the client secret of the IDP in your environment Then delete the old secret from the application.
-* If the application does not support the maintenance of multiple secrets, then create a new application. Once the new application is created, Then update the client id and client secret of the configured IDP in the environment, and . Then delete the old application.
+* If the application does not support the maintenance of multiple secrets, then create a new application. Once the new application is created, Then update the client id and client secret of the configured IDP in the environment, and .Then delete the old application.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2022-2357.md:32]:
```suggestion
* If the application does not support the maintenance of multiple secrets, then create a new application. Once the new application is created, then update the client id and client secret of the configured IDP in the environment. And then delete the old application.
```

File: en/docs/security-announcements/security-advisories/2023/WSO2-2023-2447.md [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+### SOLUTION
+We highly recommend to migrate the latest version of WSO2 Identity Server to mitigate the identified vulnerabilities.
+
+
+!!! info todo
+    **If you are a WSO2 customer with Support Subscription, please use [WSO2 Updates](https://wso2.com/updates/) in order to apply the fix.**
+
+
+### CREDITS
+WSO2 thanks, **[Tom O’Neill](https://www.linkedin.com/in/the-tom-oneill)** for responsibly reporting the identified issue and working with us as we addressed it.
```
Comment (1uffyD9) [inline en/docs/security-announcements/security-advisories/2023/WSO2-2023-2447.md:43]:
```suggestion
WSO2 thanks, [Tom O’Neill](https://www.linkedin.com/in/the-tom-oneill) for responsibly reporting the identified issue and working with us as we addressed it.
```

**Review by 1uffyD9 — CHANGES_REQUESTED** [review]
Added few suggestions.

**Review by 1uffyD9 — DISMISSED** [review]

**Comment by anjai94** [comment]
Will send the new PR.

**Pull request closed without merging**